staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware.go

```diff
@@ -227,6 +229,11 @@ func (h *UpgradeAwareHandler) ServeHTTP(w http.ResponseWriter, req *http.Request
 	if !h.UseRequestLocation {
 		newReq.URL = &loc
 	}
+	if h.OverrideRequestHost {
+		// exchanging req.Host with the backend location is necessary for backends that act on the HTTP host header (e.g. API gateways),
+		// because req.Host has preference over req.URL.Host in filling this header field
+		newReq.Host = h.Location.Host
```

Comment:
does code coverage show this line being exercised in unit tests?

Reply:
@liggitt that is covered by the new test case `use location host` in `TestServeHTTP `